doc/other/developing.rst

```diff
@@ -16,6 +16,25 @@ Phpactor which are in `lib/Extension`.
 Tests and benchmarks are maintained in the package's directory, e.g.
 `lib/WorseReflection/Tests`.
 
+How to write a code action
+--------------------------
+
+In order to implement a new code action you need the following classes:
+* CodeActionProvider
+* CodeActionCommand
+* CodeAction
+
+Let's have a look at the concept by the example of the generate decorator functionality. First off we have the
+`GenerateDecoratorProvider` this class provides a list of code actions for the file. Here we can do some pre checks e.g.
+if the file is usable to generate a decorator in. In the end the `GenerateDecoratorProvider` will return a list of
+commands that are available.
+
+The command is defined in our example in the `GenerateDecoratorCommand` which is just some glue between the actual
+code action logic and the provider.
+
+And then there is the implementation of the `WorseGenerateDecorator` which contains the logic for generating the
+decorator or more generally to apply the code action.
```

Comment:
I think this needs to be a new document rather than a section here, so `doc/development/language-server/code-action.rst`.

But contentfully this looks OK. I realise that the process for creating a code-action is quite intensive currently and could be simplified.

Reply:
I have broken up the documentation and added some more to it, about what a code action is and what kinds there are.